metadata/mapping/metadata/service_name_mapping.go

```diff
@@ -24,98 +24,81 @@ import (
 import (
 	gxset "github.com/dubbogo/gost/container/set"
 	"github.com/dubbogo/gost/log/logger"
-
-	perrors "github.com/pkg/errors"
 )
 
 import (
 	"dubbo.apache.org/dubbo-go/v3/common"
 	"dubbo.apache.org/dubbo-go/v3/common/constant"
 	"dubbo.apache.org/dubbo-go/v3/common/extension"
-	"dubbo.apache.org/dubbo-go/v3/config"
-	"dubbo.apache.org/dubbo-go/v3/config/instance"
 	"dubbo.apache.org/dubbo-go/v3/metadata/mapping"
-	"dubbo.apache.org/dubbo-go/v3/metadata/report"
-	"dubbo.apache.org/dubbo-go/v3/registry"
+	"dubbo.apache.org/dubbo-go/v3/metadata/report/instance"
 )
 
 const (
 	DefaultGroup = "mapping"
 	slash        = "/"
+	retryTimes   = 10
 )
 
 func init() {
 	extension.SetGlobalServiceNameMapping(GetNameMappingInstance)
 }
 
-// MetadataServiceNameMapping is the implementation based on metadata report
+var (
+	serviceNameMappingInstance *ServiceNameMapping
+	serviceNameMappingOnce     sync.Once
+)
+
+// GetNameMappingInstance return an instance, if not found, it creates one
+func GetNameMappingInstance() mapping.ServiceNameMapping {
+	serviceNameMappingOnce.Do(func() {
+		serviceNameMappingInstance = &ServiceNameMapping{}
+	})
+	return serviceNameMappingInstance
+}
+
+// ServiceNameMapping is the implementation based on metadata report
 // it's a singleton
-type MetadataServiceNameMapping struct {
+type ServiceNameMapping struct {
 }
 
 // Map will map the service to this application-level service
-func (d *MetadataServiceNameMapping) Map(url *common.URL) error {
+func (d *ServiceNameMapping) Map(url *common.URL) error {
 	serviceInterface := url.GetParam(constant.InterfaceKey, "")
-	// metadata service is admin service, should not be mapped
-	if constant.MetadataServiceName == serviceInterface {
-		logger.Debug("try to map the metadata service, will be ignored")
-		return nil
-	}
-
-	appName := config.GetApplicationConfig().Name
-
-	metadataReport := getMetaDataReport(url.GetParam(constant.RegistryKey, ""))
+	appName := url.GetParam(constant.ApplicationKey, "")
+	metadataReport := instance.GetMetadataReportByRegistry(url.GetParam(constant.RegistryIdKey, ""))
 	if metadataReport == nil {
-		logger.Info("get metadata report instance is nil, metadata service will be enabled!")
+		logger.Warn("get metadata report instance is nil")
```

Comment:
If metadataReport == nil, we just log and do not need to return err? It is kinda weird.